pkg/kubelet/server/server.go

```diff
@@ -105,9 +106,17 @@ func (a *filteringContainer) RegisteredHandlePaths() []string {
 }
 
 // ListenAndServeKubeletServer initializes a server to respond to HTTP network requests on the Kubelet.
-func ListenAndServeKubeletServer(host HostInterface, resourceAnalyzer stats.ResourceAnalyzer, address net.IP, port uint, tlsOptions *TLSOptions, auth AuthInterface, enableDebuggingHandlers bool) {
+func ListenAndServeKubeletServer(
+	host HostInterface,
+	resourceAnalyzer stats.ResourceAnalyzer,
+	address net.IP,
+	port uint,
+	tlsOptions *TLSOptions,
+	auth AuthInterface,
+	enableDebuggingHandlers bool,
+	runtime kubecontainer.Runtime) {
```

Comment:
I'd like to inject dependencies explicitly instead of passing in an opaque
object. I dislike the way we currently handle dependencies.

On Thu, Mar 31, 2016 at 4:08 PM, Tim St. Clair <notifications@github.com>
wrote:

> In pkg/kubelet/server/server.go
> <https://github.com/kubernetes/kubernetes/pull/23595#discussion_r58142448>
> :
>
> > @@ -105,9 +106,17 @@ func (a *filteringContainer) RegisteredHandlePaths() []string {
> >  }
> >
> >  // ListenAndServeKubeletServer initializes a server to respond to HTTP network requests on the Kubelet.
> > -func ListenAndServeKubeletServer(host HostInterface, resourceAnalyzer stats.ResourceAnalyzer, address net.IP, port uint, tlsOptions *TLSOptions, auth AuthInterface, enableDebuggingHandlers bool) {
> > +func ListenAndServeKubeletServer(
> > +	host HostInterface,
> > +	resourceAnalyzer stats.ResourceAnalyzer,
> > +	address net.IP,
> > +	port uint,
> > +	tlsOptions *TLSOptions,
> > +	auth AuthInterface,
> > +	enableDebuggingHandlers bool,
> > +	runtime kubecontainer.Runtime) {
>
> Can you just expose GetRuntime in the HostInterface instead?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23595/files/008b5221cfcb0f0297b3ff83a5f3fb27714b9eb2#r58142448>
>